docs/space-amp.md

```diff
@@ -0,0 +1,57 @@
+## Space Amp
+
+Space amp is loosely defined the total storage space used to store a database on
+media, divided by total size of the user's data.  If a database has 1000 10-byte
+keys, each with 90 byte values, then size of the user's data is 100,000 bytes.
+If the files that store this data use 131,072 of storage, then the space amp is 1.31.
+
+### Sources of Space Amp
+
+There are two sources of space amp: overhead and garbage.
+
+#### Overhead Space Amp
+
+Overhead space amp includes metadata and fragmentation.  Each storage layer
+has its own overhead, for example:
+
+- HSE overhead: mpool metadata, KVDB write ahead log, CNDB, kvset hblocks,
+  kblock headers, wbtree leaf nodes, key and value metadata.
+- Filesystem overhead: file metadata, file fragmentation.
+
+Overhead space amp is determined by HSE's on-media data structures. As a result,
+even a fully compacted KVDB will have space amp greater than 1.0.
+
+#### Garbage Space Amp
```

Comment:
I would talk about various garbage types by typical order of magnitude, so values, keys/tombs, meta, ptombs.  You should also list the fixed 32G overhead of wal buffers which appears immediately upon creation of a kvdb.  That's an enormous amount of space amp until your kvdb become really big...

Reply:
There are no WAL files on a newly created KVDB. I bet the 32G overhead is coming from the FS metadata and should be there even before a KVDB is created.

Reply:
Last I checked, the space consumption of an empty KVDB was around 32MiB or so on XFS...

Reply:
I was holding on to this PR because I wanted to add more info to the document, such as:
- HSE's overheads
- What HSE measures
- Explanation of HSE's space amp and how it relates to manually calculated space amp using `df` or `du`.

I would also like to provide detailed space amp stats via CLI so users could see how much overhead by type, how much garbage, etc.

Since there were questions flying around about space amp, I figured I'd push up the PR now and add the extra information later.